<?xml version="1.0"?>

<launch>
  <!-- Parameter Arguments -->
  <!-- mav_name is used for parameter filenames, and for the gazebo link name, Exclude dot extensions -->
  <arg name="mav_name" value="quadcopter"/>
  <arg name="waypoint_list" value="waypoints"/>

  <!-- Load  Parameters -->
  <rosparam command="load" file="$(find roscopter)/params/$(arg mav_name).yaml"/>
  <rosparam command="load" file="$(find roscopter)/params/$(arg waypoint_list).yaml"/>


  <!-- FCU communication -->
  <node name="rosflight_io" pkg="rosflight" type="rosflight_io" output="screen">
    <param name="udp" value="false"/>
    <param name="port" value="/dev/ttyACM0"/>
  </node>

  <!-- Motion Capture Node-->
  <node name="mocap" pkg="optitrack_vrpn" type="node" output="screen"/>

  <!-- PID Controller -->
  <node name="controller" pkg="roscopter" type="controller">
    <remap from="estimate" to="odom"/>
  </node>

  <!-- Estimator -->
  <node name="estimator" type="ekf_node" pkg="roscopter" output="screen">
    <param name="param_filename" value="$(find roscopter)/params/$(arg mav_name).yaml"/>
    <param name="param_namespace" value="estimator"/>
    <remap from="imu" to="/imu/data"/>
    <remap from="pose" to="/MOTIONCAPTUREASSET_ned"/>
    <!-- <remap from="range" to=<RANGE_TOPIC> /> -->
  </node>

  <!-- Waypoint Manager -->
  <node name="waypoint_manager" pkg="roscopter" type="waypoint_manager.py" output="screen">
    <remap from="waypoint" to="high_level_command"/>
    <remap from="state" to="odom"/>
  </node>

</launch>
